_posts/2024-02-07-#28950.md

```diff
@@ -8,17 +8,43 @@ authors: [instagibbs]
 components: ["rpc"]
 host: ismaelsadeeq
 status: upcoming
-commit:
+commit: bf6fbea9794db5e8d0adce8ff6182574808cea54
 ---
 
-_Notes and questions to follow soon!_
-
-<!-- TODO: Before meeting, add notes and questions
 ## Notes
+- The `submitpackage` RPC receives topologically sorted transactions, a child, and its unconfirmed parents. It validates them against consensus and policy rules, if they passes these checks, the transactions are added to the mempool.
+
+- Packages submitted to the `submitpackage` RPC are only added to the node's mempool and not relayed to peers. Package relay is currently a work in progress.
+
+- `maxfeerate` is an optional parameter for the RPCs `sendrawtransaction` and `testmempoolaccept`. Users will use it to ensure that the transaction they are adding to their mempools and broadcasting to the network does not exceed some feerate, preventing accidental payment of an unreasonable transaction fee rate.
+
+- `maxburnamount` is also an optional parameter of `sendrawtransaction` that prevents broadcasting transactions with a data carrier `OP_RETURN` output that exceeds a certain value. The default value of `maxburnamount` is 0.
+
+- The PR attempts to add optional parameters `maxfeerate` and `maxburnamount` to the `submitpackage` RPC to perform sanity checks on packages submitted to the mempool.
+
 ## Questions
+
 1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)? What was your review approach?
--->
 
+2. Why is it important to perform these sanity checks on submitted packages?
+
+3. Are there other important sanity checks apart from `maxburnamount` and `maxfeerate` that should performed on packages before they are submitted?
+
+4. The options `maxburnamount` and `maxfeerate` can prevent a transaction from entering the mempool and being relayed. Can we consider these options as policy rules?
+
+5. In [bf6fbea97](https://github.com/bitcoin/bitcoin/pull/28950/commits/bf6fbea9794db5e8d0adce8ff6182574808cea54), the commit message states, "This allows subpackage processing and is compatible with future package RBF work." How is this possible?
+
+6. What's the difference between modified feerate and absolute fee rate?
+
+7. Why are we validating against the modified feerate instead of the absolute fee rate?
+
+8. In [bf6fbea97](https://github.com/bitcoin/bitcoin/pull/28950/commits/bf6fbea9794db5e8d0adce8ff6182574808cea54), we are validating `maxfeerate` against the modified feerate of individual package transactions, not package feerate, this can sometimes be incentive incompatible. 
+    - How does this happen ?
+    - Why isn't it a current concern ? 
+    - Under what conditions might it become problematic ?
+
+
+9. Is there another approach from [bf6fbea97](https://github.com/bitcoin/bitcoin/pull/28950/commits/bf6fbea9794db5e8d0adce8ff6182574808cea54) ? What are the tradeoffs?
```

Comment:
In a single-commit PR, no need to keep linking to the same commit

Reply:
Fixed